Disable time-wasting Fastlane checks when building from CI

mokagio opened the pull request:
This could result in up to 20 minutes of extra CI time when building the WordPress iOS app.

See for example [this build](https://buildkite.com/automattic/wordpress-ios/builds/8823#0181e9c5-aa9f-4b9f-9f4d-7426c452f24a) where the `ios_build_prechecks` and `ios_build_preflight` sequence
starts [at 20:27:59](https://buildkite.com/automattic/wordpress-ios/builds/8823#0181e9c5-aa9f-4b9f-9f4d-7426c452f24a/1057-1078) and finishes [at 20:48:26](https://buildkite.com/automattic/wordpress-ios/builds/8823#0181e9c5-aa9f-4b9f-9f4d-7426c452f24a/1057-1244), 20 minutes later.

<img width="1170" alt="Screen Shot 2022-07-13 at 3 27 55 pm" src="https://user-images.githubusercontent.com/1218433/178657599-f4a2bca7-8d7f-4690-9b98-56bd21b37a6c.png">

<img width="1164" alt="image" src="https://user-images.githubusercontent.com/1218433/178657567-a98672b9-615c-4bd1-ac28-bba7e36c33ab.png">

I considered removing the call to `ios_build_preflight` altogether, but maybe there's value in having them in those rare circumstances when we run builds locally? I might open an issue to discuss this option further, but in the meantime disabling them for CI, where all those operations are already done anyways, is enough to speed up the release builds.

Here's a link to the source for `ios_build_prechecks`: https://github.com/wordpress-mobile/release-toolkit/blob/801fecfb0ed472afe70601ef75b5b523e9fed270/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_build_preflight.rb#L6-L37 to check what it does, which is

- Remove `DerivedData`
- Check `convert` and `gs`, from ImageMagick and GhostScript respectively, are present (this is only relevant for builds that update the icon, like the App Center ones)
- Removes the CocoaPods cache – This step was the time sucker, as it destroyed all the work we put into caching the pods 🤦‍♂️ 
- Run `pod install` via Fastlane

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

mokagio committed:
Disable time wasting Fastlane checks when building from CI

This could result in up to 20 minutes of extra CI time when building the
WordPress iOS app.

See for example the build at
https://buildkite.com/automattic/wordpress-ios/builds/8823#0181e9c5-aa9f-4b9f-9f4d-7426c452f24a,
where the `ios_build_prechecks` and `ios_build_preflight` sequence
starts at 20:27:59
(https://buildkite.com/automattic/wordpress-ios/builds/8823#0181e9c5-aa9f-4b9f-9f4d-7426c452f24a/1057-1078)
and finishes at 20:48:26, 20 minutes later
(https://buildkite.com/automattic/wordpress-ios/builds/8823#0181e9c5-aa9f-4b9f-9f4d-7426c452f24a/1057-1244)

AliSoftware commented:
> Removes the CocoaPods cache – This step was the time sucker, as it destroyed all the work we put into caching the pods 🤦‍♂️

IMHO we should actually remove this `Action.sh('rake dependencies:pod:clean')` from the release-toolkit's action altogether (for the next release we'll do of release-toolkit).

AliSoftware commented on .buildkite/commands/release-build-wordpress-internal.sh:
```diff
@@ -20,4 +20,6 @@ echo "--- :closed_lock_with_key: Installing Secrets"
-bundle exec fastlane build_and_upload_app_center skip_confirm:true
+bundle exec fastlane build_and_upload_app_center
+  skip_confirm:true \
+  skip_precheks:true
```
```suggestion
  skip_prechecks:true
```

AliSoftware commented on .buildkite/commands/release-build-wordpress.sh:
```diff
@@ -22,5 +22,6 @@ bundle exec fastlane run configure_apply
 echo "--- :hammer_and_wrench: Building"
 bundle exec fastlane build_and_upload_app_store_connect \
   skip_confirm:true \
+  skip_precheks:true \
```
```suggestion
  skip_prechecks:true \
```

AliSoftware reviewed (dismissed):
Typo in the param name.

Also, should we make this target the `release/` branch, so that you also get the speedup benefit ASAP for the next betas and releases that you'll do this sprint?

mokagio commented:
@AliSoftware 

> we make this target the `release/` branch, so that you also get the speedup benefit ASAP for the next betas and releases that you'll do this sprint?

Yep. I forgot to do that. Good catch. On the param name, too.

mokagio commented:
I rebased and force pushed after changing the base branch. I addressed the typo fix (🤦‍♂️) by squashing onto the first commit while rebasing. Since I was already rebasing, I thought I'd make the commit atomic.

mokagio commented:
Mindful of the typo from the original iteration here, I setup a test PR for the change this time: https://github.com/wordpress-mobile/WordPress-iOS/pull/19046...

...which was a good idea because, while one build worked, the other one didn't 🤔 

![image](https://user-images.githubusercontent.com/1218433/178977622-f6204d90-a54f-4e9b-a5dc-ba6254e0ab65.png)

<img width="1155" alt="image" src="https://user-images.githubusercontent.com/1218433/178977691-ef71a49c-fd7b-438c-859b-482d82e9b93f.png">

I'll need to look more into this.

AliSoftware commented on .buildkite/commands/release-build-wordpress-internal.sh:
```diff
@@ -20,4 +20,6 @@ echo "--- :closed_lock_with_key: Installing Secrets"
 
 echo "--- :hammer_and_wrench: Building"
-bundle exec fastlane build_and_upload_app_center skip_confirm:true
+bundle exec fastlane build_and_upload_app_center
```
> while one build worked, the other one didn't 🤔

Here's your mistake: you forgot the `\` at the end of the line so that the lines below would be considered the continuation of this one. That's why the CI failed, because the `skip_confirm:true` (and subsequent params) were not actually passed to the lane due to that missing `\` here.
```suggestion
bundle exec fastlane build_and_upload_app_center \
```

mokagio committed:
Add missing `\` for multi-line command in script for App Center build

mokagio committed:
Use emojis for WordPress and Jetpack in `release-builds` pipeline

It sort of upsets me how the step names gets truncated in the build
pipeline visualization and one can't tell which box is for App Center
and which for App Store Connect.

I added a custom emoji for Jetpack in Buildkite,
https://github.com/buildkite/emojis/pull/392, and now's the time to use
it.

mokagio committed:
Use custom emojis for WordPress and Jetpack in release pipeline (#19052)

mokagio commented:
@crazytonyli can I trouble you with a review of this?

Unfortunately, I messed up the base merges and https://github.com/wordpress-mobile/WordPress-iOS/pull/19052 went into this branch, even though it's unrelated.

You can see a demo of the change in effect [here](https://buildkite.com/automattic/wordpress-ios/builds/9189), based on [this test](https://github.com/wordpress-mobile/WordPress-iOS/commit/47537ae9e2cbb7f19f4124243a33bbac410a7894). Notice the logs in the two successful WordPress builds. I cancelled the Jetpack one because unrelated with these changes (it did not reset the CocoaPods cache).

<img width="1061" alt="image" src="https://user-images.githubusercontent.com/1218433/180898234-e58d1fa6-2970-4b68-b544-28ae7a639cd5.png">


Thanks!
